x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDecider.java

```diff
@@ -192,7 +185,12 @@ private Decision shouldIndexPreferTier(IndexMetadata indexMetadata, Set<Discover
     }
 
     public static String[] parseTierList(String tiers) {
-        return Strings.tokenizeToStringArray(tiers, ",");
+        if (Strings.hasLength(tiers) == false) {
```

Comment:
To avoid an array like `[" "]` (not that we expect that), maybe we should use:
```suggestion
        if (Strings.hasText(tiers) == false) {
```